app/controllers/works_controller.rb

```diff
@@ -0,0 +1,91 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @works = Work.all
+  end
+
+  def show
+    if @work.nil?
+      redirect_to works_path
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+
+    if @work.save
+      flash[:success] = "Successfully created #{@work.category}"
+      redirect_to work_path(@work.id)
+      return
+    else
+      flash.now[:danger] = "Failed to create work"
+      render :new, status: :bad_request
+      return
+    end
+  end
+
+  def edit
+    if @work.nil?
+      redirect_to works_path
+      return
+    end
+  end
+
+  def update
+    if @work.nil?
+      head :not_found
+      return
+    elsif @work.update(work_params)
+      flash[:success] = "Successfully updated #{@work.category}"
+      redirect_to work_path(@work.id)
+      return
+    else
+      flash.now[:danger] = "Failed to update work"
+      render :edit, status: :bad_request
+      return
+    end
+  end
+
+  def destroy
+    if @work
+      @work.destroy
+      flash[:success] = "Successfully deleted #{@work.category}"
+      redirect_to works_path
+      return
+    else
+      head :not_found
+      return
```

Comment:
At this point, avoid using `head` instead use redirect or render a nice 404 page.